v2/internal/reconcilers/entra/entra_securitygroup_reconciler.go

```diff
@@ -0,0 +1,499 @@
+		}
+
+		return ctrl.Result{}, eris.Wrapf(err, "failed to delete group %q (%s)", displayName, id)
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *EntraSecurityGroupReconciler) Claim(
+	ctx context.Context,
+	log logr.Logger,
+	eventRecorder record.EventRecorder,
+	obj genruntime.MetaObject,
+) error {
+	// Nothing to do
+	//!! Confirm this is true
+	return nil
+}
+
+// update completes our reconciliation by updating an existing Entra security group.
+// ctx is the context for the operation.
+// id is the Entra ID of the group to update.
+// group is the security group to update.
+// eventRecorder is used to record events for the group.
+// log is the logger to use for logging.
+func (r *EntraSecurityGroupReconciler) update(
+	ctx context.Context,
+	id string,
+	group *asoentra.SecurityGroup,
+	log logr.Logger,
+) (ctrl.Result, error) {
+	log = log.WithValues("id", id)
+	log.V(Status).Info("Updating Entra security group")
+
+	// Create our Entra Client
+	client, err := r.EntraClientFactory(ctx, group)
+	if err != nil {
+		return ctrl.Result{}, eris.Wrap(err, "creating entra client prior to update")
+	}
+
+	// Load the existing group by ID
+	g, err := r.loadGroupByID(ctx, id, client.Client())
+	if err != nil {
+		if r.isNotFound(err) {
+			// Group used to exist, but no longer does - it's probably been deleted
+			// Remove the existing annotation and requeue the reconciliation to create a replacement
+			log.V(Status).Info("Group no longer exists")
+			setEntraID(group, "")
+			return ctrl.Result{
+				Requeue: true,
+			}, nil
+		}
+
+		return ctrl.Result{}, eris.Wrapf(err, "getting group by ID %s", id)
+	}
+
+	// Update - PATCH
+	group.Spec.AssignToGroup(g)
+
+	_, err = client.Client().Groups().ByGroupId(id).Patch(ctx, g, nil)
+	if err != nil {
+		// Failed to update
+		return ctrl.Result{}, eris.Wrapf(err, "failed to update group %s", id)
+	}
+
+	group.Status.AssignFromGroup(g)
+
+	return ctrl.Result{}, nil
+}
+
+// tryAdopt tries to find an existing Entra security group to adopt.
+// ctx is the context for the operation.
+// id is the Entra ID of the group to update.
+// group is the security group to update.
+// eventRecorder is used to record events for the group.
+// log is the logger to use for logging.
+// Returns the Entra ID of the group to adopt, and nil if found;
+// nil, nil if no group was found to adopt;
+// or nil, error if there was a problem finding the group.
+func (r *EntraSecurityGroupReconciler) tryAdopt(
+	ctx context.Context,
+	group *asoentra.SecurityGroup,
+	log logr.Logger,
+) (*string, error) {
+	log.V(Status).Info("Searching for existing Entra security group to adopt", "group", group.Name)
+
+	// Create our Entra Client
+	client, err := r.EntraClientFactory(ctx, group)
+	if err != nil {
+		return nil, eris.Wrap(err, "creating entra client prior to adoption search")
+	}
+
+	// Try to find the group by DisplayName
+	displayName := group.Spec.DisplayName
+	if displayName == nil || *displayName == "" {
```

Comment:
very minor, style: could be `to.FromPtr(displayName) == ""` I think

Reply:
Or above: `to.FromPtr(group.Spec.DisplayName)`

Reply:
Package `"github.com/Azure/azure-service-operator/v2/internal/util/to"` doesn't have a FromPtr method. Leaving the way it is.

Reply:
Fixed.